test/ectest.c

```diff
@@ -836,8 +802,7 @@ static int char2_curve_test(int n)
         || !TEST_true(BN_hex2bn(&p, test->p))
         || !TEST_true(BN_hex2bn(&a, test->a))
         || !TEST_true(BN_hex2bn(&b, test->b))
-        || !TEST_true(group = EC_GROUP_new(EC_GF2m_simple_method()))
-        || !TEST_true(EC_GROUP_set_curve(group, p, a, b, ctx))
```

Comment:
Sidenote: do you have thoughts about deprecating also EC_GROUP_set_curve? (as another PR in case it makes sense)

Reply:
To elaborate more:

Once `EC_GROUP_new(some_method)` function is deprecated, I don't see leftover reasons to set parameters after instantiation.

Reply:
I agree in the sense that ... the only time I've written `EC_GROUP_set_curve` in code is exactly the file you're looking at. To be able to reuse an `EC_GROUP` without re-initializing it.

But outside of a test, I don't see why anyone would do that.

That is ofc just my subjective opinion -- I have not checked for applications using `EC_GROUP_set_curve`.

Reply:
Maybe we can add this item to the parent issue you were using to track all these related PRs?
We can try to do an analysis of that and see if deprecation can be recommended.

Reply:
I added `EC_GROUP_set_curve` to the #8332 list for analysis.